pkg/volume/csi/csi_client.go

```diff
@@ -177,10 +178,30 @@ func (c *csiDriverClient) NodeGetInfo(ctx context.Context) (
 	accessibleTopology map[string]string,
 	err error) {
 	klog.V(4).Info(log("calling NodeGetInfo rpc"))
-	if c.nodeV1ClientCreator != nil {
-		return c.nodeGetInfoV1(ctx)
-	} else if c.nodeV0ClientCreator != nil {
-		return c.nodeGetInfoV0(ctx)
+
+	backoff := gensupport.DefaultBackoffStrategy()
+	var pause time.Duration
+
+	// Exponential backoff
+	for {
+		select {
+		case <-ctx.Done():
+			return nodeID, maxVolumePerNode, accessibleTopology, ctx.Err()
+		case <-time.After(pause):
+		}
+
+		if c.nodeV1ClientCreator != nil {
+			return c.nodeGetInfoV1(ctx)
```

Comment:
Is there a way to differentiate between terminal and non-terminal errors (so that we don't retry needlessly)?